src/wavm/mpi.cpp

```diff
@@ -1223,6 +1223,8 @@ WAVM_DEFINE_INTRINSIC_FUNCTION(env,
 /**
  * Makes a new communicator to which Cartesian topology information has been
  * attached.
+ * Note: In OpenMPI, memory is allocated from within the function call, that's
+ * why we allocate it here.
```

Comment:
It's not just because OpenMPI does it, it's because that's the MPI interface right?